config/config.go

```diff
@@ -268,6 +276,25 @@ func mergeConfigWithIncludedConfig(config *TerragruntConfig, includedConfig *Ter
 	return includedConfig, nil
 }
 
+// Merge the extra arguments prioritizing those defined in the childExtraArgs
+func mergeExtraArgs(terragruntOptions *options.TerragruntOptions, childExtraArgs []TerraformExtraArguments, parentExtraArgs *[]TerraformExtraArguments) {
+	result := *parentExtraArgs
+addExtra:
+	for _, child := range childExtraArgs {
+		for i, parent := range result {
+			if parent.Name == child.Name {
+				terragruntOptions.Logger.Printf("extra_arguments '%v' from child overriding parent", child.Name)
+				// For extra args, we want to keep the values specified in the child and put them after
+				// the parent ones, so if we encounter a duplicate, we just overwrite it.
+				result[i] = child
+				continue addExtra
```

Comment:
Agreed.  I just took @jocgir 's code and ran with it.  

However, now that I think about it, I think we do want to impose some ordering here (which a map can't provide).  For example, if the parent extra_arguments includes a variable file containing `variable_x`, and a child also includes a variable file containing `variable_x`, then I hope we can order the extra_arguments such that the child's file "wins".  I'm not sure if terraform honors var file ordering on the cli, or which one "wins" (first file containing the variable, or last file containing the variable).  I'll do a little investigation.